### PR TITLE
Refresh tower-http

### PR DESCRIPTION
**Pull request by carllerche**
This is a work in progress to bring `tower-http` up to date.

### CODE REVIEW

File: tower-http-service/src/lib.rs [hunk 1]
```diff
@@ -33,37 +44,20 @@ pub trait HttpService<RequestBody>: Sealed<RequestBody> {
     /// `HttpService` instance needs to be used where a `T: Service` is
     /// required, it must be wrapped with a type that provides that
     /// implementation. `LiftService` does this.
-    fn lift(self) -> LiftService<Self> where Self: Sized {
-        LiftService { inner: self }
+    fn into_service(self) -> util::IntoService<Self> where Self: Sized {
```
Comment (LucioFranco) [inline tower-http-service/src/lib.rs]: why not just import these instead of using the full path?

File: tower-http-service/src/util/as_service.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+use HttpService;
+use futures::Poll;
+use http::{Request, Response};
+use tower_service::Service;
+
+/// Wraps an `HttpService` reference, implementing `tower_service::Service`.
+///
+/// See [`lift`] function documentation for more details.
```
Comment (LucioFranco) [inline tower-http-service/src/util/as_service.rs]: lift no longer exists?

Comment (LucioFranco) [reply tower-http-service/src/util/as_service.rs]: prob should be `AsService` instead

File: tower-http-service/src/lib.rs (hunk 1 above)
Comment (carllerche) [reply tower-http-service/src/lib.rs]: 🤷‍♂️

File: Cargo.toml [hunk 3]
```diff
@@ -1,16 +1,5 @@
-[package]
-name = "tower-http"
```
Comment (GeorgeHahn) [inline Cargo.toml]: I think this broke the `tower-grpc` build - that project still contains references to this package.

Comment (carllerche) [reply Cargo.toml]: Yep... I messed that up. Fixes are coming.

Comment (carllerche) [reply Cargo.toml]: The recent churn is an effort to get the whole thing on crates.io :) so sorry about that.

File: src/service.rs [hunk 4]
```diff
@@ -1,123 +0,0 @@
...
-        LiftServiceRef { inner: self }
-    }
-}
-
-/// Wraps an `HttpService` instance, implementing `tower_service::Service`.
-///
-/// See [`lift`] function documentation for more details.
-///
-/// [`lift`]: #
-pub struct LiftService<T> {
```
Comment (hntd187) [inline src/service.rs]: Some downstream services like @LucioFranco 's tower-consul breaks with lift service being removed, lift should just be replaced with straight service impls correct?

Comment (LucioFranco) [reply src/service.rs]: lift just got renamed to `AsService` so just need to change the functions we call in tower-consul

Comment (hntd187) [reply src/service.rs]: There is more to it than that, LiftService only has T, where AsService is `AsService<'a, T>`

Comment (carllerche) [reply src/service.rs]: * `LiftService` -> `IntoService`
* `LiftServiceRef` -> `AsService`.

It should have just been a rename. If not, I messed up.

**Pull request merged**